Update the README to include additional information on tags and best practices (#72)
smartdevicelink/sdl_app_library_guides: README.md

# SDL App Library Guide Content

> iOS and Java Suite (Android, JavaEE, and JavaSE) guide content for the [SDL Developer Portal](https://smartdevicelink.com/en/guides/iOS/getting-started/installation/).

Reviewer: ```suggestion
The iOS and Java Suite (Android, JavaEE, and JavaSE) guides are located on the [SDL Developer Portal](https://smartdevicelink.com/en/guides/iOS/getting-started/installation/).
```

## Overview

> Documents in this guide are created in a custom SDL markdown format. All guides get created as directories inside this repository. Each directory gets a link in the left-hand sidebar on [smartdevicelink.com](https://smartdevicelink.com). Guides do not currently support multiple pages or sub-directories. Guides specific to one platform or with content specific to one platform will use custom tags to denote this content.

Reviewer: ```suggestion
Since both the iOS and Java Suite guides provide almost the same instructions and information, the content of each section should be written as generically as possible. This allows the the guides to be written once for all platforms with little modification needed. If a guide is specific to one platform or has content specific to one platform, like code snippets, the content should be separated out using custom platform tags.

Guide documents are formatted using a custom SDL markdown format. All guides get created as directories inside this repository. Each directory gets a link in the left-hand sidebar on [smartdevicelink.com](https://smartdevicelink.com). Guides do not currently support multiple pages or sub-directories
```

Learn how to use SDL markdown (aka DocDown) at the [spec page](https://github.com/smartdevicelink/sdl_markdown_spec).

## Custom SDL Markdown Tags

> Basic markdown tags will [as expected](https://github.com/smartdevicelink/sdl_markdown_spec#basic-markdown-syntax) including headings, bold, italics, and tables. However, there are [custom extensions](https://github.com/smartdevicelink/sdl_markdown_spec#basic-markdown-syntax) for inline notes and relative links to assets.

Reviewer: ```suggestion
Basic markdown tags will act [as expected](https://github.com/smartdevicelink/sdl_markdown_spec#basic-markdown-syntax), including headings, bold, italics, and tables. In addition, there are [SDL specific custom extensions](https://github.com/smartdevicelink/sdl_markdown_spec#basic-markdown-syntax) for inline notes and relative links to assets.
```

### Specifying Platforms

> This repository has additional customizations to permit content to be included or excluded from one of the several platforms of documentation that this repository builds. This can be done inline within a guide in order to, for example, display one code snippet for iOS and another for Java-family platforms. This can also be done within the configuration file in order to allow a guide such as deploying a Java Web app to AWS to only build for the JavaEE platform.

Reviewer: ```suggestion
This repository has additional markdown customizations to permit content to be included or excluded from a platform. This can be done inline within a guide in order to, for example, display one code snippet for iOS and another for Java-family platforms. This can also be done within the configuration file in order to allow a guide, such as a guide demoing how to deploy a Java Web app to AWS, to only build for the JavaEE platform.
```

#### Valid Tags

| Platform         | Tag                    |
| ---------------- | ---------------------- |
| iOS              | @![iOS] ------ !@      |
| Android          | @![android] -- !@      |
| JavaEE           | @![javaEE] --- !@      |
| JavaSE           | @![javaSE] --- !@      |

#### How to Use Inline Tags
Inline tags allow you to create content that is only rendered for the specified platforms within a guide that is available on multiple platforms. For example, if you have a guide available on Android, JavaSE, and JavaEE but not on iOS (see below How to Use Configuration Tags for how to configure guide availability), then you can use inline tags to render some content for Android that won't be rendered for JavaSE or JavaEE, or visa versa.

    # How To Code Stuff
    Here is non-platform specific documentation and instructions that tell the developer things and stuff and how to do whatever
    
    ## Code Example
    @![android]

>     This is Android specific instructions that only applies to the Android > platform specifically.

Reviewer: ```suggestion
    This is Android specific instructions that only applies to the Android platform.
```

    
    ```java
    (Android code snippet)
    ``` !@
    

>     @![javaSE,javaEE]

Reviewer: ```suggestion
    This is JavaSE and JavaEE specific instructions that only applies to the Java platform.
    @![javaSE,javaEE]
```

    ```java
    (Java code snippet)
    ``` !@
    
#### How to Use Configuration Tags

> Configuration tags allow to you create guides that are only available for the specified platforms.

Reviewer: ```suggestion
Configuration tags allow to you create guides that are only available for the specified platforms. If no platforms are included for a section, the section will be generated for all possible configured platforms. 
```
